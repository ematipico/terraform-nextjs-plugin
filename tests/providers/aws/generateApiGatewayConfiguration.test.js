// @ts-nocheck
const { generateTerraformConfiguration } = require("../../../src/providers/aws/generateApiGatewayConfiguration");
const { setConfiguration } = require("../../../src/configuration");
const path = require("path");

describe("generateTerraformConfiguration", () => {
	it("should generate the correct configuration", async () => {
		setConfiguration({
			gatewayKey: "CustomKey",
			lambdaPath: path.resolve(__dirname, "__fixtures__"),
			routes: {
				prefix: "",
				mappings: [
					{
						page: "/blog-post",
						route: "/new-blog-post-amazing"
					},
					{
						page: "/singleBlogPost",
						route: "/blog/:url"
					},
					{
						page: "/blogPost",
						route: "/blog/detail/:url",
						params: {
							page: true,
							hideComments: false
						}
					}
				]
			}
		});

		const result = await generateTerraformConfiguration();

		expect(result).toMatchSnapshot();
	});
});
