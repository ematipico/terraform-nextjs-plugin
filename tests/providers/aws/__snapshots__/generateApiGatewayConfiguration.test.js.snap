// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateTerraformConfiguration should generate the correct configuration 1`] = `
Object {
  "resource": Object {
    "aws_api_gateway_integration": Object {
      "CustomKey-blog-detail-url": Object {
        "http_method": "GET",
        "integration_http_method": "POST",
        "request_parameters": Object {
          "integration.request.path.url": "method.request.path.url",
          "integration.request.querystring.hideComments": "method.request.querystring.hideComments",
          "integration.request.querystring.page": "method.request.querystring.page",
        },
        "resource_id": "\${aws_api_gateway_resource.CustomKey-blog-detail-url.id}",
        "rest_api_id": "\${aws_api_gateway_rest_api.CustomKey.id}",
        "type": "AWS_PROXY",
        "uri": "arn:aws:apigateway:\${local.aws_region}:lambda:path/2015-03-31/functions/\${aws_lambda_function.lambdaForCustomKey-blogPost.arn}/invocations",
      },
      "CustomKey-blog-url": Object {
        "http_method": "GET",
        "integration_http_method": "POST",
        "request_parameters": Object {
          "integration.request.path.url": "method.request.path.url",
        },
        "resource_id": "\${aws_api_gateway_resource.CustomKey-blog-url.id}",
        "rest_api_id": "\${aws_api_gateway_rest_api.CustomKey.id}",
        "type": "AWS_PROXY",
        "uri": "arn:aws:apigateway:\${local.aws_region}:lambda:path/2015-03-31/functions/\${aws_lambda_function.lambdaForCustomKey-singleBlogPost.arn}/invocations",
      },
      "CustomKey-boar": Object {
        "http_method": "GET",
        "integration_http_method": "POST",
        "resource_id": "\${aws_api_gateway_resource.CustomKey-boar.id}",
        "rest_api_id": "\${aws_api_gateway_rest_api.CustomKey.id}",
        "type": "AWS_PROXY",
        "uri": "arn:aws:apigateway:\${local.aws_region}:lambda:path/2015-03-31/functions/\${aws_lambda_function.lambdaForCustomKey-boar.arn}/invocations",
      },
      "CustomKey-contact-us": Object {
        "http_method": "GET",
        "integration_http_method": "POST",
        "resource_id": "\${aws_api_gateway_resource.CustomKey-contact-us.id}",
        "rest_api_id": "\${aws_api_gateway_rest_api.CustomKey.id}",
        "type": "AWS_PROXY",
        "uri": "arn:aws:apigateway:\${local.aws_region}:lambda:path/2015-03-31/functions/\${aws_lambda_function.lambdaForCustomKey-contact-us.arn}/invocations",
      },
      "CustomKey-index": Object {
        "http_method": "GET",
        "integration_http_method": "POST",
        "resource_id": "\${aws_api_gateway_resource.CustomKey-index.id}",
        "rest_api_id": "\${aws_api_gateway_rest_api.CustomKey.id}",
        "type": "AWS_PROXY",
        "uri": "arn:aws:apigateway:\${local.aws_region}:lambda:path/2015-03-31/functions/\${aws_lambda_function.lambdaForCustomKey-index.arn}/invocations",
      },
      "CustomKey-new-blog-post-amazing": Object {
        "http_method": "GET",
        "integration_http_method": "POST",
        "resource_id": "\${aws_api_gateway_resource.CustomKey-new-blog-post-amazing.id}",
        "rest_api_id": "\${aws_api_gateway_rest_api.CustomKey.id}",
        "type": "AWS_PROXY",
        "uri": "arn:aws:apigateway:\${local.aws_region}:lambda:path/2015-03-31/functions/\${aws_lambda_function.lambdaForCustomKey-blog-post.arn}/invocations",
      },
    },
    "aws_api_gateway_method": Object {
      "CustomKey-blog-detail-url": Object {
        "authorization": "NONE",
        "http_method": "GET",
        "request_parameters": Object {
          "method.request.path.url": true,
          "method.request.querystring.hideComments": false,
          "method.request.querystring.page": true,
        },
        "resource_id": "\${aws_api_gateway_resource.CustomKey-blog-detail-url.id}",
        "rest_api_id": "\${aws_api_gateway_rest_api.CustomKey.id}",
      },
      "CustomKey-blog-url": Object {
        "authorization": "NONE",
        "http_method": "GET",
        "request_parameters": Object {
          "method.request.path.url": true,
        },
        "resource_id": "\${aws_api_gateway_resource.CustomKey-blog-url.id}",
        "rest_api_id": "\${aws_api_gateway_rest_api.CustomKey.id}",
      },
      "CustomKey-boar": Object {
        "authorization": "NONE",
        "http_method": "GET",
        "resource_id": "\${aws_api_gateway_resource.CustomKey-boar.id}",
        "rest_api_id": "\${aws_api_gateway_rest_api.CustomKey.id}",
      },
      "CustomKey-contact-us": Object {
        "authorization": "NONE",
        "http_method": "GET",
        "resource_id": "\${aws_api_gateway_resource.CustomKey-contact-us.id}",
        "rest_api_id": "\${aws_api_gateway_rest_api.CustomKey.id}",
      },
      "CustomKey-index": Object {
        "authorization": "NONE",
        "http_method": "GET",
        "resource_id": "\${aws_api_gateway_resource.CustomKey-index.id}",
        "rest_api_id": "\${aws_api_gateway_rest_api.CustomKey.id}",
      },
      "CustomKey-new-blog-post-amazing": Object {
        "authorization": "NONE",
        "http_method": "GET",
        "resource_id": "\${aws_api_gateway_resource.CustomKey-new-blog-post-amazing.id}",
        "rest_api_id": "\${aws_api_gateway_rest_api.CustomKey.id}",
      },
    },
    "aws_api_gateway_resource": Object {
      "CustomKey-blog": Object {
        "parent_id": "\${aws_api_gateway_rest_api.CustomKey.root_resource_id}",
        "path_part": "blog",
        "rest_api_id": "\${aws_api_gateway_rest_api.CustomKey.id}",
      },
      "CustomKey-blog-detail": Object {
        "parent_id": "\${aws_api_gateway_resource.CustomKey-blog.id}",
        "path_part": "detail",
        "rest_api_id": "\${aws_api_gateway_rest_api.CustomKey.id}",
      },
      "CustomKey-blog-detail-url": Object {
        "parent_id": "\${aws_api_gateway_resource.CustomKey-blog-detail.id}",
        "path_part": "{url}",
        "rest_api_id": "\${aws_api_gateway_rest_api.CustomKey.id}",
      },
      "CustomKey-blog-url": Object {
        "parent_id": "\${aws_api_gateway_resource.CustomKey-blog.id}",
        "path_part": "{url}",
        "rest_api_id": "\${aws_api_gateway_rest_api.CustomKey.id}",
      },
      "CustomKey-boar": Object {
        "parent_id": "\${aws_api_gateway_rest_api.CustomKey.root_resource_id}",
        "path_part": "boar",
        "rest_api_id": "\${aws_api_gateway_rest_api.CustomKey.id}",
      },
      "CustomKey-contact-us": Object {
        "parent_id": "\${aws_api_gateway_rest_api.CustomKey.root_resource_id}",
        "path_part": "contact-us",
        "rest_api_id": "\${aws_api_gateway_rest_api.CustomKey.id}",
      },
      "CustomKey-index": Object {
        "parent_id": "\${aws_api_gateway_rest_api.CustomKey.root_resource_id}",
        "path_part": "index",
        "rest_api_id": "\${aws_api_gateway_rest_api.CustomKey.id}",
      },
      "CustomKey-new-blog-post-amazing": Object {
        "parent_id": "\${aws_api_gateway_rest_api.CustomKey.root_resource_id}",
        "path_part": "new-blog-post-amazing",
        "rest_api_id": "\${aws_api_gateway_rest_api.CustomKey.id}",
      },
    },
  },
  "variable": Object {
    "integrationList": Object {
      "default": Array [
        "aws_api_gateway_integration.CustomKey-new-blog-post-amazing",
        "aws_api_gateway_integration.CustomKey-blog-url",
        "aws_api_gateway_integration.CustomKey-blog-detail-url",
        "aws_api_gateway_integration.CustomKey-boar",
        "aws_api_gateway_integration.CustomKey-contact-us",
        "aws_api_gateway_integration.CustomKey-index",
      ],
    },
  },
}
`;
